awx_collection/plugins/modules/tower_credential.py

```diff
@@ -257,162 +274,117 @@
 )
 
 
-def credential_type_for_kind(params):
-    credential_type_res = tower_cli.get_resource('credential_type')
-    kind = params.get('kind')
-    arguments = {'managed_by_tower': True}
-    if kind == 'ssh':
-        if params.get('vault_password'):
-            arguments['kind'] = 'vault'
-        else:
-            arguments['kind'] = 'ssh'
-    elif kind in ('net', 'scm', 'insights', 'vault'):
-        arguments['kind'] = kind
-    elif kind in KIND_CHOICES:
-        arguments.update(dict(
-            kind='cloud',
-            name=KIND_CHOICES[kind]
-        ))
-    return credential_type_res.get(**arguments)
-
-
 def main():
-
+    # Any additional arguments that are not fields of the item can be added here
     argument_spec = dict(
         name=dict(required=True),
+        new_name=dict(),
+        description=dict(),
+        organization=dict(),
+        credential_type=dict(),
+        inputs=dict(type='dict'),
         user=dict(),
         team=dict(),
+        # These are for backwards compatability
         kind=dict(choices=list(KIND_CHOICES.keys())),
-        credential_type=dict(),
-        inputs=dict(type='dict'),
         host=dict(),
         username=dict(),
         password=dict(no_log=True),
-        ssh_key_data=dict(no_log=True, type='str'),
+        project=dict(),
+        ssh_key_data=dict(no_log=True),
         ssh_key_unlock=dict(no_log=True),
-        authorize=dict(type='bool', default=False),
+        authorize=dict(type='bool'),
         authorize_password=dict(no_log=True),
         client=dict(),
         security_token=dict(),
-        secret=dict(),
-        tenant=dict(),
+        secret=dict(no_log=True),
         subscription=dict(),
+        tenant=dict(),
         domain=dict(),
         become_method=dict(),
         become_username=dict(),
         become_password=dict(no_log=True),
         vault_password=dict(no_log=True),
-        description=dict(),
-        organization=dict(required=True),
-        project=dict(),
-        state=dict(choices=['present', 'absent'], default='present'),
         vault_id=dict(),
+        # End backwards compatability
+        state=dict(choices=['present', 'absent'], default='present'),
     )
 
-    mutually_exclusive = [
-        ('kind', 'credential_type')
```

Comment:
I'm hitting an impasse at the test that checked for this. The parameters for the task there were

```python
        name='A credential',
        organization=organization.name,
        kind='ssh',
        credential_type=cred_type.name,
        state='present'
```

I can't get on board with allowing this task to run. Is the credential type ssh, or is it the custom cred_type?

I feel like the nicest thing we can do is fail. This is such a massive contradiction in parameters, no one should keep this in their playbooks.

Reply:
It would be the custom credential type. We do call this out the modules behavior in the documentation:
```
    credential_type:
      description:
        - Name of credential type.
        - Will be preferred over kind
      required: False
      version_added: "2.10"
      type: str
```
But we can put the mutually exclusive option back in if you'd prefer setting both to cause an error.

Reply:
Update to tests pushed.